xgcm/grid.py

```diff
@@ -1255,16 +1255,16 @@ def check_neighbor(link, position):
             self.axes[axis]._facedim = facedim
             self.axes[axis]._connections = axis_links
 
-    def _assign_metrics(self, metrics):
-        """
-        metrics should look like
-           {('X', 'Y'): ['rAC']}
-        check to make sure everything is a valid dimension
-        """
-
-        self._metrics = {}
+    _metrics = {}
```

Comment:
really minor comment: it's convention to move these kinds of variables up above `def __init__`

Reply:
So I should replace `metrics=None`?

```python
class Grid:
    """
    An object with multiple :class:`xgcm.Axis` objects representing different
    independent axes.
    """

    def __init__(
        self,
        ds,
        check_dims=True,
        periodic=True,
        default_shifts={},
        face_connections=None,
        coords=None,
         _metrics = {},
        boundary=None,
        fill_value=None,
    ):
```

Reply:
sorry for the confusion, I meant
```
class Grid:
	_metrics = {}

	def __init__(
		...,
		metrics=None,
		...,
	):
		...
```

Reply:
Oh I see, thanks!

Reply:
So I should retain `metrics=None` under def __init__?

Reply:
yes that way users can opt-in to providing metrics...

Reply:
@dcherian, we might want to be careful here so as to avoid race conditions and other weird outcomes. Using `_metrics` as a class attribute instead of an instance attribute results into unintended outcomes:


```python
n [1]: class Grid:
   ...:     _metrics = {}
   ...: 

In [2]: a = Grid()

In [3]: a._metrics['key'] = 'value'

In [5]: a._metrics
Out[5]: {'key': 'value'}

In [6]: b = Grid()

In [7]: b._metrics
Out[7]: {'key': 'value'}
```

Should we make `_metrics` an instance attribute instead?

Reply:
Oh oops! thanks Anderson

``` python
class Grid:

	def __init__(
		...,
		metrics=None,
		...,
	):
		self._metrics = {}

		# parse metrics dict here
		...
```

Reply:
I just saw that `_metrics` is an instance attribute in the existing version. Are there any benefits to making this a class attribute?

Reply:
No, I was typing without thinking.